io/input/inputformat_factory_unittest.cpp

```diff
@@ -0,0 +1,174 @@
+    EXPECT_EQ(&infmt2, &infmt4);
+
+    // Drop
+    InputFormatFactory::drop_inputformat("infmt1");
+    EXPECT_EQ(InputFormatFactory::size(), 1);
+    InputFormatFactory::drop_inputformat("infmt2");
+    EXPECT_EQ(InputFormatFactory::size(), 0);
+}
+
+TEST_F(TestInputFormatFactory, CreateBinaryInputFormat) {
+    // Create
+    auto& infmt1 = InputFormatFactory::create_binary_inputformat("nfs:///dummy", "", "infmt1");
+    EXPECT_EQ(InputFormatFactory::size(), 1);
+    auto& infmt2 = InputFormatFactory::create_binary_inputformat("nfs:///dummy", "", "infmt2");
+    EXPECT_EQ(InputFormatFactory::size(), 2);
+
+    // Get
+    auto& infmt3 = InputFormatFactory::get_binary_inputformat("infmt1");
+    EXPECT_EQ(&infmt1, &infmt3);
+    auto& infmt4 = InputFormatFactory::get_binary_inputformat("infmt2");
+    EXPECT_EQ(&infmt2, &infmt4);
+
+    // Drop
+    InputFormatFactory::drop_inputformat("infmt1");
+    EXPECT_EQ(InputFormatFactory::size(), 1);
+    InputFormatFactory::drop_inputformat("infmt2");
+    EXPECT_EQ(InputFormatFactory::size(), 0);
+}
+
+#ifdef WITH_THRIFT
+TEST_F(TestInputFormatFactory, CreateThriftInputFormat) {
+    // Create
+    auto& infmt1 = InputFormatFactory::create_thrift_inputformat("localhost", 2016, "infmt1");
+    EXPECT_EQ(InputFormatFactory::size(), 1);
+    auto& infmt2 = InputFormatFactory::create_thrift_inputformat("localhost", 2016, "infmt2");
+    EXPECT_EQ(InputFormatFactory::size(), 2);
+
+    // Get
+    auto& infmt3 = InputFormatFactory::get_thrift_inputformat("infmt1");
+    EXPECT_EQ(&infmt1, &infmt3);
+    auto& infmt4 = InputFormatFactory::get_thrift_inputformat("infmt2");
+    EXPECT_EQ(&infmt2, &infmt4);
+
+    // Drop
+    InputFormatFactory::drop_inputformat("infmt1");
+    EXPECT_EQ(InputFormatFactory::size(), 1);
+    InputFormatFactory::drop_inputformat("infmt2");
+    EXPECT_EQ(InputFormatFactory::size(), 0);
+}
+#endif
+
+#ifdef WITH_MONGODB
+TEST_F(TestInputFormatFactory, CreateMongoDBInputFormat) {
+    Context::init_global();
+    // Create
+    auto& infmt1 = InputFormatFactory::create_mongodb_inputformat("infmt1");
+    EXPECT_EQ(InputFormatFactory::size(), 1);
+    auto& infmt2 = InputFormatFactory::create_mongodb_inputformat("infmt2");
+    EXPECT_EQ(InputFormatFactory::size(), 2);
+
+    // Get
+    auto& infmt3 = InputFormatFactory::get_mongodb_inputformat("infmt1");
+    EXPECT_EQ(&infmt1, &infmt3);
+    auto& infmt4 = InputFormatFactory::get_mongodb_inputformat("infmt2");
+    EXPECT_EQ(&infmt2, &infmt4);
+
+    // Drop
+    InputFormatFactory::drop_inputformat("infmt1");
+    EXPECT_EQ(InputFormatFactory::size(), 1);
+    InputFormatFactory::drop_inputformat("infmt2");
+    EXPECT_EQ(InputFormatFactory::size(), 0);
+Context::finalize_local();
+Context::finalize_global();
+}
+#endif
+
+// InputFormatFactory will register a finalizer in SessionLocal
+TEST_F(TestInputFormatFactory, DummyFinalizer) {
+    husky::base::SessionLocal::get_thread_finalizers().clear();
```

Comment:
Actually it's not a test. If this is necessary, maybe it can be added in `TearDown` of the test class.